src/main/java/mate/academy/service/impl/ReportImpl.java

```diff
@@ -0,0 +1,28 @@
+package mate.academy.service.impl;
+
+import java.util.Map;
+import mate.academy.dao.FruitDao;
+import mate.academy.dao.FruitDaoImpl;
+import mate.academy.service.ReportGenerator;
+
+public class ReportImpl implements ReportGenerator {
+    private static final String REPORT_HEADER = "fruit,quantity";
+    private static final String DATA_SEPARATOR = ",";
+    private final FruitDao fruitDao;
+
+    public ReportImpl() {
+        fruitDao = new FruitDaoImpl();
+    }
```

Comment:
The code should not be implementation dependent, fix everywhere
```suggestion
    public ReportImpl(FruitDao fruitDao) {
        this.fruitDao = fruitDao;
    }
```